src/Components/ProductListing/ProductListing.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState } from "react";
+import useCommerce from "../commerce-context/commerce-context";
+import { useReducer } from "react";
+import "./productListing.css";
+
+import sortReducer from "./sortReducer";
+import filterReducer from "./filterReducer";
+import ProductsDisplay from "./ProductsDisplay";
+
+import ProductsFilters from "./ProductsFilters";
+import ProductsSorting from "./ProductsSorting";
+import ProductsSearch from "./ProductsSearch";
+
+export default function ProductListingPage({ setRoute }) {
+  const { state } = useCommerce();
+  const [filterBttn, setFilterBttn] = useState("Hide");
+
+  const [sortData, sortDispatch] = useReducer(sortReducer, {
+    products: []
+  });
+
+  useEffect(
+    () =>
+      sortDispatch({ type: "DATA_FROM_CONTEXT", payload: state.ProductsList }),
+    [state]
+  );
+
+  const [
+    { showOnlyInStock, showFastDelivery, priceRange, searchProducts },
+    filterDispatch
+  ] = useReducer(filterReducer, {
+    showOnlyInStock: true,
+    showFastDelivery: false,
+    priceRange: 1000,
+    searchProducts: ""
+  });
+
+  function filterData(

```

Comment:
Can put these functions in their own respectives files in a folder named "helpers" or "utils"? Idk, just a thought.

Reply:
sure will do